Create CustomRefreshIndicator - edited for upgrade

thomaskim1130 opened the pull request:
## Overview
<!-- 간단하게 이 PR이 무엇인지 설명해주세요. 예: 새로운 로그인 기능 추가 -->
로딩을 위해 pull down하는 거리 감소. 편의성 증진
## Changes
<!-- 이 PR로 인해 변경되는 사항을 나열해주세요. 예:
- 로그인 페이지 UI 업데이트
- 로그인 API 연동
- 로그인 에러 처리 추가
-->
- refreshIndicator.adaptive()를 ScrollNotification 클래스를 활용해 custom화 함. 
- 기존 pull down 50(변경 불가) 에서 30으로 조정. (변경 가능)
- main_page.dart, post_list_show_page.dart, user_page.dart, post_view_page 등 페이지에 적용
## Implementaion Method
<!-- 변경사항을 구현한 방법에 대해 특별히 전달해야할 것이 있다면 설명해주세요. 예:
- React Hook을 사용하여 상태 관리
- JWT를 사용한 인증 처리
-->
- utils/refresh_indicator.dart를 추가하여 메소드 customRefreshIndicator 생성
## After Changes
<!-- 변경 사항을 확인 할 수 있는 방법을 알려주세요. 가능하다면, 변화를 보여주는 스크린샷 또는 동영상을 첨부해주세요. 예:
- 로그인 성공 시 홈페이지로 리다이렉트하는 동영상
-->
- scrollDownLength를 customize할 수 있음
- radius, color, 애니메이션 딜레이 등도 모두 customize 가능
- refresh 함수를 호출하는 중 중복되지 않음
## Related Issues
<!-- 관련 버그 현상이나 관련 이슈 번호가 있다면 링크를 적어주거나 스크린샷을 첨부해주세요. 예: #123 -->
- LoadingIndicator(화면 중앙)와 동시에 나타나는 현상이 발생하기도 함
## Rollback Scenario
<!-- 해당 PR를 롤백하는 방법과 순서를 알려주세요. 예: 
1. revert commit
2. .env 파일 수정
-->
1. customRefreshIndicator 대신 refreshIndicator.adaptive 사용
2. AlwaysScrollablePhysics의 parent 변경하기
## TODO
<!-- 앞으로 해야하는 것이나, 논의가 필요한 부분 등을 적어주세요. 예:
- 로그인 관련 테스트 케이스 추가
- 다국어 지원 검토
-->
- user_view_page.dart 에도 적용. (DONE)
- refreshIndicator 다수 중복. Wrapper 함수 제작? (DONE)
- DarkMode 적용

thomaskim1130 committed:
Add(refresh_indicator.dart): created new customRefreshIndicator

Created new CustomRefreshIndicator that:
- doesn't rely on any global keys
- thus doesn't cause state issues
- based on actual animation + NotificationListener
- can customize radius, pull down length, offset, etc.
- based on CupertinoActionIndicator

thomaskim1130 committed:
Feat: applied customRefreshIndicator

Applied customRefreshIndicator widget to :
mainPage
notificationPage
postListShowPage
postViewPage
userPage
userViewPage

(also modified physics for android)

thomaskim1130 committed:
Fix(main_page.dart): fixed

added back asynchronous operation

thomaskim1130 commented:
> ## Overview
> 로딩을 위해 pull down하는 거리 감소. 편의성 증진
> 
> ## Changes
> * refreshIndicator.adaptive()를 ScrollNotification 클래스를 활용해 custom화 함.
> * 기존 pull down 50(변경 불가) 에서 30으로 조정. (변경 가능)
> * main_page.dart, post_list_show_page.dart, user_page.dart, post_view_page 등 페이지에 적용
> 
> ## Implementaion Method
> * utils/refresh_indicator.dart를 추가하여 메소드 customRefreshIndicator 생성
> 
> ## After Changes
> * scrollDownLength를 customize할 수 있음
> * radius, color, 애니메이션 딜레이 등도 모두 customize 가능
> * onRefresh 함수를 호출하는 중 중복되지 않음
> 
> ## Related Issues
> * LoadingIndicator(화면 중앙)와 동시에 나타나는 현상이 발생하기도 함
> 
> ## Rollback Scenario
> 1. customRefreshIndicator 대신 refreshIndicator.adaptive 사용
> 2. AlwaysScrollablePhysics의 parent 변경하기
> 
> ## TODO
> * user_view_page.dart 에도 적용. (DONE)
> * refreshIndicator 다수 중복. Wrapper 함수 제작? (DONE)
> * Dark Mode 적용

sangohkim commented on lib/widgets/refresh_indicator.dart:
```diff
@@ -0,0 +1,209 @@
+const scrollDownLength = 30.0; // refresh를 위한 최소 픽셀
+const displacementLength = 0.0; // 위로부터의 default displacement
+const offsetLength = 5.0; // 위/아래의 default offset
+const indicatorRadius = 15.0; // 원의 default radius
```
Flutter에서 사용하는 기본값(_kDefaultIndicatorRadius) 10.0으로 변경 부탁드립니다

sangohkim reviewed (changes requested):
RefreshIndicator radius 관련한 간단한 수정 요청 드렸습니다. 시뮬레이터에서 본 결과 전체적으로 잘 동작하는것 같습니다! 안드로이드에서는 CustomRefreshIndicator만 cupertino 스타일이고 나머지는 material 디자인에 맞춰져있어서 부자연스러울 수도 있겠다는 생각이 드는데 CustomRefreshIndicator를 iOS 일 때만 적용해보는건 어떠신가요?

thomaskim1130 committed:
Feat(refresh_indicator.dart): multi platform compatibility

Applied CustomRefreshIndicator ONLY for iOS (Cupertino), replaced with Material for Android.
Reset default radius to 10.0

thomaskim1130 committed:
Merge branch 'dev' into feature/change-refresh-pulldown-edit

thomaskim1130 committed:
Fix(theme_info.dart): Color bug fix

scroll 시 그림자 발생하는 Bug fix, AppBarTheme을 ThemeData 내부에 도입하면서, scrolledUnderElevation=0.0으로 설정하여 해결

sangohkim reviewed (changes requested):
<img width="951" alt="image" src="https://github.com/user-attachments/assets/1b060176-279f-495a-bc9f-2e090d4f286e" />

```flutter run``하고 나서 홈화면으로 들어가면 위 에러가 발생하여 수정 요청드립니다.
기종은 iphone 16 (iOS 18.1), Pixel (Android API 34) 입니다

thomaskim1130 committed:
Fix(connectivity_provider.dart): Fixed false internet error

Added ConnectivityResult.mobile (LTE) to the conditions for not triggering the internet error.

sparcscasio reviewed (approved):
LGTM,, 늦게 확인해서 죄송합니다ㅜㅠㅜㅠ

thomaskim1130 reviewed (commented):
해결했습니다

sangohkim reviewed (approved):
LGTM
